lib/datasets_test.go

```diff
@@ -439,21 +515,48 @@ func TestDatasetRequestsGetFSIPath(t *testing.T) {
 		t.Fatalf("error checking out dataset: %s", err)
 	}
 
-	getParams := &GetParams{
-		Ref: "peer/movies",
-	}
-	got, err := inst.Dataset().Get(ctx, getParams)
-	if err != nil {
-		t.Fatalf("error getting fsi dataset: %s", err)
-	}
-	if got.Ref.Username != "peer" {
-		t.Errorf("incorrect Username, expected 'peer', got %q", got.Ref.Username)
-	}
-	if got.Ref.Name != "movies" {
-		t.Errorf("incorrect Username, expected 'movies', got %q", got.Ref.Name)
+	var nilCommit *dataset.Commit
+	fsiExpectedDs := dataset.Dataset{}
+	fsiExpectedDs.Assign(moviesDs)
+	fsiExpectedDs.Stats = nil
+	fsiExpectedDs.Commit = nil
+
+	cases := []struct {
+		description string
+		params      *GetParams
+		expect      interface{}
+	}{
+
+		{"ref without path",
+			&GetParams{Ref: "peer/movies"},
+			setDatasetName(&fsiExpectedDs, "peer/movies")},
+
```

Comment:
This test case is failing, and TBH I'm, not exactly sure what should be "correct" in this case.

I know when we "checkout" a dataset to fsi, we don't pull computer generated fields into the file, since these should not be editable.

However, for this get function, should we only be passing through the content of the files, or should we be passing through a completed look at the dataset? Basically, should fields like "commit", "stats", and "structure.checksum" be empty, or should they be filled in with the content from the latest committed version?

I don't understand the expectations/uses, so I'm having trouble answering this question.

Here is the diff:
```              &dataset.Dataset{
              	... // 1 ignored field
              	Body:      nil,
              	BodyBytes: nil,
            - 	BodyPath:  "/mem/QmQpAK5bPJKj65Bmzwa8WuoASiLUW4Hf4hR4h8ddPnEmAe",
            + 	BodyPath:  "/var/folders/4d/4pgbh8d1083g1f7skny7bwy00000gn/T/get_fsi_test120231655/movies/body.csv",
              	Commit:    nil,
              	ID:        "",
              	Meta: &dataset.Meta{
              		... // 1 ignored and 9 identical fields
              		Language:  nil,
              		License:   nil,
            - 		Path:      "/mem/QmZQNhYYVRx8LyMmPV9mqzVZVEeZKpso4Ywu7nwyWvT4X4",
            + 		Path:      "",
              		Qri:       "md:0",
              		ReadmeURL: "",
              		... // 3 identical fields
              	},
              	Name:         "movies",
            - 	Path:         "/mem/QmV6bPoBtAg2P3v1cTLjyXjcaYKodjwiYhywxzo8h36m9d",
            + 	Path:         "",
              	Peername:     "peer",
              	PreviousPath: "",
              	... // 2 identical fields
              	NumVersions: 0,
              	Qri:         "",
              	Structure: &dataset.Structure{
            - 		Checksum:     "/mem/QmQpAK5bPJKj65Bmzwa8WuoASiLUW4Hf4hR4h8ddPnEmAe",
            + 		Checksum:     "",
              		Compression:  "",
            - 		Depth:        2,
            + 		Depth:        0,
              		Encoding:     "",
            - 		ErrCount:     4,
            + 		ErrCount:     0,
            - 		Entries:      2335,
            + 		Entries:      0,
              		Format:       "csv",
              		FormatConfig: {"headerRow": bool(true)},
            - 		Length:       49753,
            + 		Length:       0,
            - 		Path:         "/mem/QmQqNmcUdpt9x6cUYp4YJNpppkQGhLEDzs1PsNtzQmi1RK",
            + 		Path:         "",
              		Qri:          "st:0",
              		Schema:       {"items": map[string]interface{}{"items": []interface{}{map[string]interface{}{"title": string("title"), "type": string("string")}, map[string]interface{}{"title": string("duration"), "type": string("integer")}}, "type": string("array")}, "type": string("array")},
              		Strict:       false,
              	},
              	Stats:     nil,
              	Transform: nil,
              	Viz:       nil,
              }
```

Reply:
So yes, the current implementation has been that `Get` on a dataset which has a working directory will only ever look at the files on disk, in that working directory. It won't load anything from the repository or IPFS. As you rightfully point out, this means that the `stats` and `commit` will always be nil, since we don't store any "machine-targeted" files for an FSI directory. The structure is similar; a structure in a working directory will only contain the `Format`, `FormatConfig` and `Schema` fields, since the rest of the fields are all values derived from the last save and so shouldn't be user-editable.

This is arguably a misfeature. It means that commands like `qri get commit.message` won't work in FSI; so it's not possible to view the most recent commit information. I would argue that we should fix that, but the fix is probably not appropriate for this refactoring, and if it's not too hard we should keep the current behavior. Fixing the larger issue probably needs a bit of discussion and should be its own separate PR. An earlier issue talking about this problem: https://github.com/qri-io/qri/issues/1507

This is also definitely a good example of behavior we should document with the long discussed spec.

Reply:
This is extremely helpful, thank you.